Utils/FaviconFetcher.swift

```diff
@@ -128,8 +128,16 @@ public class FaviconFetcher : NSObject, NSXMLParserDelegate {
                                 iconType = nil
                         }
                     }
+
+                    guard let href = link.attribute("href") where iconType != nil else {
+                        return
```

Comment:
I'm reading it wrong. Missed that we were in a function rather than a loop. Ignore my idiocy.